### DISCUSSION TITLE
“Missing Database Password” in VM Logs

### DISCUSSION THREAD

**Discussion by AvoMandjian** [serverpod/serverpod#2601] [Q&A] [answered]
Hello,

I am attempting to host on fly.io and setting secrets or environment variables, but I continue to receive the following error in the VM logs: `Exception: Missing database password.`

These are the variables I have tried to set:

1. `password`
2. `databasePassword`
3. `database`
4. `TF_VAR_DATABASE_PASSWORD_PRODUCTION`
5. `SERVERPOD_DATABASE_PASSWORD`
6. `POSTGRES_PASSWORD`

Could you please confirm if one of these is the correct variable, or if I should be using a different one?

Thank you.

**Comment by SandPod** [comment, marked as answer]
Serverpod 2.0.2 does not support loading the database password from an environment variable.
But it will be supported in the next version of Serverpod, for which we will release a beta soon.

In the meanwhile, you can read the supported keywords and the documentation here: https://docs.serverpod.dev/next/concepts/configuration

**Comment by iampopal** [comment]
Any updates about steps on how to deploy to fly.io?

**Comment by iampopal** [comment]
I am also getting error of: 

Exception: Missing database password.

Any steps to solve that.

**Reply by SandPod** [reply]
If you are using at least Serverpod 2.1.0 you can set the database password using the `SERVERPOD_DATABASE_PASSWORD` env variable.

**Reply by SandPod** [reply]
Instructions on how to set env variable can be found here for fly.io: https://fly.io/docs/rails/the-basics/configuration/

**Reply by iampopal** [reply]
Thank you for your replay, this variable is already set but still show error of: Exception: Missing database password.

**Reply by SandPod** [reply]
Hmm, for debugging purposes, what happens if you read the environment variable in the run method of your server?
Do you send in any configuration with the construction of your server? If so, these might override the default read of environment variable.

**Reply by iampopal** [reply]
At the end I didn't able to run it in fly.io even fly has a very good cli and quickly learnable. It will be awesome to write a blog for steps or a create a video tutorial on How to deploy to fly.io

**Reply by SandPod** [reply]
There might actually still exist i bug here, this PR fixes an issue where the envs would not be loaded if the password file was missing: https://github.com/serverpod/serverpod/pull/2801
